src/test/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecutionTest.java

```diff
@@ -241,6 +241,53 @@ private static void assertStepExecutions(FlowExecution e, String... steps) throw
         });
     }
 
+    @Issue("JENKINS-34256")
+    @Test public void quietDownThenCancelQuietDown() {
+        story.then(r -> {
+            WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+            p.setDefinition(new CpsFlowDefinition("semaphore 'wait'; echo 'I am done'", true));
+            WorkflowRun b = p.scheduleBuild2(0).waitForStart();
+            SemaphoreStep.waitForStart("wait/1", b);
+            r.jenkins.doQuietDown(true, 0);
+            SemaphoreStep.success("wait/1", null);
+            ((CpsFlowExecution) b.getExecution()).waitForSuspension();
+            assertTrue(b.isBuilding());
+            r.assertLogNotContains("I am done", b);
+            r.jenkins.doCancelQuietDown();
+            r.assertLogContains("I am done", r.assertBuildStatusSuccess(r.waitForCompletion(b)));
+        });
+    }
+
+    @Issue("JENKINS-34256")
+    @Test public void pauseAndQuietDown() {
```

Comment:
Not sure about this test, probably would need to be tweaked to avoid race conditions. The goal was to test that canceling quiet mode doesn't also unpause Pipelines.

Reply:
Removed the test. It was hard to follow, and I couldn't think of a good way to make it more robust without inspecting internal execution state in loops with sleep.